[feat/#8] 검색 결과 화면 컴포넌트 구현

vahkjsdf opened the pull request:
## ✅ 𝗖𝗵𝗲𝗰𝗸-𝗟𝗶𝘀𝘁
- 리뷰는 PR이 올라오면 최대한 빠르게 진행합니다.
- P1 단계의 리뷰는 빠르게 확인 후 반영합니다.
- Approve된 PR은 assigner가 머지하고,
  수정 요청이 온 경우 수정 후 다시 push를 합니다.

## 📌 𝗜𝘀𝘀𝘂𝗲𝘀
- closed #8

## 📎 𝗪𝗼𝗿𝗸 𝗗𝗲𝘀𝗰𝗿𝗶𝗽𝘁𝗶𝗼𝗻
- 검색 결과 화면 컴포넌트를 구현했습니다.

## 📷 𝗦𝗰𝗿𝗲𝗲𝗻𝘀𝗵𝗼𝘁
- 타이틀
   <img width="259" height="183" alt="image" src="https://github.com/user-attachments/assets/7afae547-3263-4645-a6e9-f090ee1ceb5a" />
- 도서 아이템
   <img width="221" height="512" alt="image" src="https://github.com/user-attachments/assets/5e8bf6c1-2baf-4d69-86a2-f2e748a19375" />
- 검색바
   <img width="603" height="122" alt="image" src="https://github.com/user-attachments/assets/b1dba29b-d54b-45b6-aafe-ff8b64cf33c7" />
- 서재 아이템
   <img width="220" height="386" alt="image" src="https://github.com/user-attachments/assets/9519f0f4-1570-403a-bb30-40762d2e43b0" />


## 💬 𝗧𝗼 𝗥𝗲𝘃𝗶𝗲𝘄𝗲𝗿𝘀

vahkjsdf committed:
[feat/#8] 타이틀 텍스트 구현

vahkjsdf committed:
[feat/#8] 책 제목, 저자 텍스트 구현

vahkjsdf committed:
[add/#8] wandok icon 추가

vahkjsdf committed:
[feat/#8] 완독률 텍스트 추가

vahkjsdf committed:
[feat/#8] 오디오북 마크 구현

vahkjsdf committed:
[mod/#8] 완독률 아이콘 Image → Icon 으로 변경

vahkjsdf committed:
[feat/#8] 배너 구현

vahkjsdf committed:
[mod/#8] lint format

vahkjsdf committed:
[chore/#8] 변수명 변경

vahkjsdf committed:
[feat/#8] 도서아이템 구현

vahkjsdf committed:
[mod/#8] reader maxLine 추가 및 padding 조정

vahkjsdf committed:
[mod/#8] icon size 지정

vahkjsdf committed:
[chore/#8] preview 수정

vahkjsdf committed:
[feat/#8] 검색바 구현

vahkjsdf committed:
[chore/#8] 함수명 변경

vahkjsdf committed:
[chore/#8] 패키지 이동

vahkjsdf committed:
[feat/#8] 서재 아이템 구현

vahkjsdf committed:
[fix/#8] 충돌 해결

nhyeonii commented on app/src/main/java/sopt/org/millie/core/designsystem/component/SearchTextField.kt:
```diff
@@ -0,0 +1,104 @@
+            onSearch = {
+                onSearchAction()
+                keyboardController?.hide()
+            },
```
onDone과 onSearch가 동일한 동작을 하고 있네용 !! 위에서 `keyboardOptions = ImeAction.Search`로 사용하면 onSearch만 트리거되므로 onDone은 불필요할 것 같아요잉 ~~

vahkjsdf replied:
아 맞네용 삭제하겠습니다!

nhyeonii commented on app/src/main/java/sopt/org/millie/core/designsystem/component/SearchTextField.kt:
```diff
@@ -0,0 +1,104 @@
+            focusedContainerColor = MillieTheme.colors.lightGray1,
+            unfocusedContainerColor = MillieTheme.colors.lightGray1,
+            focusedIndicatorColor = Color.Transparent,
+            unfocusedIndicatorColor = Color.Transparent,
+        ),
+    )
+}
+
+@Preview(showBackground = true)
+@Composable
+private fun SearchTextFieldPreview() {
+    SearchTextField(
+        value = "",
+        onValueChange = {},
+    )
+}
```
```suggestion
@Preview(showBackground = true)
@Composable
private fun SearchTextFieldPreview() {
    MillieTheme {
        var text by remember { mutableStateOf("") }

        SearchTextField(
            value = text,
            onValueChange = { text = it },
        )
    }
}
```
요거 MillieTheme{} 로 안감싸주면 컬러랑 타이포를 찾을수가 없어서 프리뷰 오류날 가능성이 클텐데 ...!?!?! 그리구 interactionmode에서 타자 쳐지는거 확인해보려면 이렇게 작성하는건 어떨까요 ~ 히히

vahkjsdf replied:
헉 이런 방법이... mainactivity에 몰래 추가해서 실행해봤어요....ㅎㅎ

요렇게 수정해볼게요!!

sohee6989 replied:
귀엽다....ㅎ.ㅎ

nhyeonii commented on app/src/main/java/sopt/org/millie/presentation/search/component/book/SearchAudioTag.kt:
```diff
@@ -0,0 +1,61 @@
+                color = MillieTheme.colors.milliePurple,
+                style = MillieTheme.typography.body.caption4,
+            )
+        }
```
요거 Row로 만들수도 잇답니다 ~ !! 그리구 제안하자면 요거 뱃지 부분 컴포넌트로 따로 빼는거 오때요 ??! SearchAudioTag는 검색 결과에서 오디오북 정보를 표시하는 역할이고, 뱃지 자체는 독립적인 UI 컴포넌트로 볼 수도 있을 것 같아용 ㅎㅎ 지미니 생각이 궁금해요 !!

vahkjsdf replied:
뱃지를 컴포넌트로 빼면 어디까지 컴포넌트로 빼야할지 감이 안와서..ㅜ 혹시 다른 화면에서 이 뱃지 컴포넌트를 사용할 일이 있나요?

nhyeonii replied:
<img width="288" height="139" alt="image" src="https://github.com/user-attachments/assets/53d3d827-bf92-41b6-8919-35346c99cabd" />

하게 되면 요 Box부분만 빼면 되겟죵 ?!? 우리 세미나 때 배웠던 것처럼 다른 곳에서 사용되지 않더라도 가독성을 높이기 위해서 컴포넌트화 하는 것도 좋은 방법이에요 !

nhyeonii commented on app/src/main/java/sopt/org/millie/presentation/search/component/book/SearchAudioTag.kt:
```diff
@@ -0,0 +1,61 @@
+            text = reader,
+            modifier = Modifier
+                .padding(start = 4.dp)
+                .weight(1f),
```
요거 텍스트에 .weight(1f) 이 속성 준 이유가 잇나요 ?!? 맨 최상위 Row의 modifier 속성에 fillMaxWidth()써도 같은 효과를 줄 수 있어요 !! 자식의 weight(1f) 대신 부모의 fillMaxWidth()를 사용하는 게 더 간결하고 명확할 것 같은데 어떤가용 ?!

vahkjsdf replied:
fillMaxWidth()랑 같은 효과를 주고 싶어서 사용했어요..! 코드 작성할 때는 저 속성이 생각이 안나서 weight로 대체했었네용.. 수정하겠습니다!

nhyeonii commented on app/src/main/java/sopt/org/millie/presentation/search/component/book/SearchBookItem.kt:
```diff
@@ -0,0 +1,65 @@
+
+        SearchAudioTag(
+            reader = reader,
+        )
```
지민아 우리 오디오북 태그 없는 것도 있어서 이거 분기처리하는 코드가 필요할 것 같아요 !!
<img width="320" height="359" alt="image" src="https://github.com/user-attachments/assets/17922ef5-cc62-4df1-a002-4ba397d6af90" />
현재는 다 오디오북 있는 애들로만 구현될 것 같으네 !!

vahkjsdf replied:
맞다.. 추가할게용ㅎㅎ

nhyeonii commented on app/src/main/java/sopt/org/millie/presentation/search/component/book/SearchBookItem.kt:
```diff
@@ -0,0 +1,65 @@
+) {
+    Column(
+        modifier = modifier.width(103.dp),
+        verticalArrangement = Arrangement.spacedBy(1.dp),
```
<img width="454" height="370" alt="image" src="https://github.com/user-attachments/assets/aab8d391-a9fc-42fc-bbd8-bc19cbedbca7" />

요기는 간격 2.dp라 계산해서 반영해줘야할것 같아용 !!

vahkjsdf replied:
헉 잘못봤나보네요,, 수정하겠습니다!

nhyeonii commented on app/src/main/java/sopt/org/millie/presentation/search/component/book/SearchWandokRate.kt:
```diff
@@ -0,0 +1,60 @@
+) {
+    Row(
+        modifier = modifier,
+        horizontalArrangement = Arrangement.spacedBy(4.dp),
```
<img width="460" height="230" alt="image" src="https://github.com/user-attachments/assets/98ee3b49-0ae4-42f4-a010-c5d3afc9d7cf" />
요기는 간격이 2.dp라 이렇게 주면 전부 4.dp씩 떨어지는거라 수정 필요할것 같소 ㅜㅜ...

vahkjsdf replied:
넵ㅜ!

nhyeonii commented on app/src/main/java/sopt/org/millie/presentation/search/component/book/SearchWandokRate.kt:
```diff
@@ -0,0 +1,60 @@
+            tint = MillieTheme.colors.gray3,
+        )
+        Text(
+            text = rate,
```
아직 서버 명세가 안나와서 모르겟는데 혹시 서버에서 rate를 %안붙이구 Int로 준다면 ` text = "${rate}%",` 요렇게 작성해주면 대겟죠 ~?

vahkjsdf replied:
명세 보니까 이 부분 수정해야할 것 같아요! 수정해서 올릴게용

nhyeonii commented on app/src/main/java/sopt/org/millie/presentation/search/component/book/SearchWandokRate.kt:
```diff
@@ -0,0 +1,60 @@
+            style = MillieTheme.typography.body.body1,
+        )
+        Text(
+            text = minute,
```
여기두 ~~ 마찬가지 ! 아직 명세가 안나와서 수정할 필요는 업서요 !!

nhyeonii reviewed (commented):
와 머야 !! 왜케 잘해 !! 텍스트필드 어려운건데 넘 구현 잘해줘서 반햇다 .. 🫰🫰 자잘하게 수정하면 좋을 부분들 있어서 리뷰 남겨둿어용 ~~ 한번씩 확인해보면 넘 좋을 것 같아요 진짜 잘햇다 진짜 !!!!!

nhyeonii commented on app/src/main/java/sopt/org/millie/presentation/search/component/library/LibraryItem.kt:
```diff
@@ -0,0 +1,61 @@
+
+@Composable
+fun LibraryItem(
+    imgRes: Int,
```
요거 @DrawableRes 어노테이션 붙여줍시다 !!! 나두 잊고 잇엇다

vahkjsdf replied:
오 알겠습니당

nhyeonii commented on app/src/main/java/sopt/org/millie/presentation/search/component/book/SearchBookItem.kt:
```diff
@@ -0,0 +1,65 @@
+    modifier: Modifier = Modifier,
+) {
+    Column(
+        modifier = modifier.width(103.dp),
```
요거 그리구 클릭했을 때 책 상세 페이지로 이동해야해서 clickable 속성도 추가해줘야할 것 같앙요 ~~

vahkjsdf replied:
넹 추가할게요!

sohee6989 commented on app/src/main/java/sopt/org/millie/presentation/search/component/library/LibraryItem.kt:
```diff
@@ -0,0 +1,61 @@
+            contentDescription = null,
+            modifier = Modifier
+                .clip(CircleShape)
+                .size(65.dp, 66.dp),
```
여기 size에서도 width, height 파라미터명 명시해주면 더 가독성이 좋을 것 같아요!

sohee6989 commented on app/src/main/java/sopt/org/millie/presentation/search/component/book/SearchWandokRate.kt:
```diff
@@ -0,0 +1,60 @@
+            contentDescription = null,
+            modifier = Modifier.size(15.dp),
+            tint = MillieTheme.colors.gray3,
+        )
```
컴포넌트 사이에 한 줄 간격 주면 가독성이 더 좋을 것 같아요!

sohee6989 commented on app/src/main/java/sopt/org/millie/presentation/search/component/book/SearchBanner.kt:
```diff
@@ -0,0 +1,40 @@
+    Box(
+        modifier = modifier
+            .width(311.dp)
+            .height(84.dp)
```
여기 지금 가로, 세로를 고정값으로 줬는데요!
기기대응 이런걸 생각해봤을때 여기에서 고정값 대신 fillMaxWidth로 주고 
이걸 사용하는 화면에서 가로 패딩으로 이 컴포넌트의 크기를 지정해주는게 더 좋을 것 같아요!

vahkjsdf replied:
기기 대응..! 생각지도 못했는데 너무 좋습니다👍

sohee6989 commented on app/src/main/java/sopt/org/millie/presentation/search/component/book/SearchBookItem.kt:
```diff
@@ -0,0 +1,65 @@
+            modifier = Modifier
+                .fillMaxWidth()
+                .height(150.dp)
+                .padding(bottom = 12.dp),
```
요기 나중에 제꺼 머지되고 나면 풀 당겨와서 shadow 관련 확장함수 써서 shadow 줘야 될 것 같아요~!!

vahkjsdf replied:
넹!!

sohee6989 commented on app/src/main/java/sopt/org/millie/presentation/search/component/book/SearchBookItem.kt:
```diff
@@ -0,0 +1,65 @@
+) {
+    Column(
+        modifier = modifier.width(103.dp),
+        verticalArrangement = Arrangement.spacedBy(1.dp),
```
추가로 지금 이미지랑 책 제목 사이 간격들 반영 되어 있나요??
없는 것 같은데 혹시 있는데 제가 놓친건가 싶기도 하고..!ㅎㅎ

vahkjsdf replied:
이미지랑 책 제목 사이 간격이 13.dp인데 컴포넌트 간의 간격이 1.dp 여서 spacedBy로 1.dp씩 준 후에 이미지 아래에 패딩을 12.dp 주었습니다! 근데 전체적으로 패딩을 다시 확인해서 수정해야할 것 같아요..ㅎㅎ

sohee6989 commented on app/src/main/java/sopt/org/millie/core/designsystem/component/SearchTextField.kt:
```diff
@@ -0,0 +1,104 @@
+                        ),
+                )
+            } else {
+                trailingIcon()
```
단순 궁금한 점이 있어서 여쭤봅니다...!
지금 검색어에 입력 내용이 있을 때 없을 때 분기처리는 잘 해놓으셨는데요!
값이 없을 때의 아이콘은 여기서 지정 안 하고 trainglinIcon으로 따로 상위에서 넣도록 처리하신 이유가 궁금합니다!

vahkjsdf replied:
이 검색바는 홈화면이랑 검색 결과화면에서 같이 사용하는 컴포넌트입니다!
홈화면에서는 입력 내용이 없을 때 바코드 아이콘이 들어가는데 검색 결과 화면에서는 아무것도 들어가지 않아서 상위에서 넣도록 처리했습니당

nhyeonii replied:
요건 지민님 대신 답변 드리자면 .. !! 현재 SearchTextField는 홈화면과 검색 화면에서 사용되고 있는데요, 빈 상태일때의 trailing icon이 화면 마다 다를 수 있기 때문이에용 ...!! 홈 화면을 보면 검색바 끝 부분에 바코드 스캔 아이콘이 필요하고, 검색 화면에서는 아예 아이콘이 없어야하죠 !! 반면에 입력값이 있을 때 삭제 버튼을 보여주는 것은 검색 필드의 공통적인 UX 패턴이기 때문에 컴포넌트의 내부에서 처리하는 것이 더 적합할 거에요 !! 만약 빈 상태의 아이콘도 내부에서 고정해버리면 화면마다 다른 요구 사항이 생길 때 컴포넌트를 수정하거나 아예 새로 만들어야하는 문제가 생길거에요 !! 그래서 trailingIcon을 상위에서 처리하도록 구현해두엇을 거에요 !!

vahkjsdf replied:
넵 맞아요! 
입력 내용이 없을 때의 아이콘은 각 화면에서 처리하는 것이 맞다고 생각해서 상위에서 넣도록 처리했습니다ㅎㅎ

sohee6989 replied:
아아 이해했습니당~!!
감사합니다아아~~❤️

sohee6989 commented on app/src/main/java/sopt/org/millie/core/designsystem/component/SearchTextField.kt:
```diff
@@ -0,0 +1,104 @@
+                    modifier = Modifier
+                        .size(24.dp)
+                        .noRippleClickable(
+                            onClick = { onValueChange("") },
```
이 부분 삭제 관련 처리를 상위로 호이스팅해서 관리해보는 건 어떠세요?

```suggestion
                            onClick = onCancelClick,
```

vahkjsdf replied:
오 좋아요!

sohee6989 reviewed (approved):
수고하셨습니다!!👍👍
특히 텍필 컴포넌트는 저도 항상 만들때마다 어려운데 너무 잘하신것 같아요!!

vahkjsdf committed:
[mod/#8] 코드 리뷰 수정

- 검색바 취소 상태 호이스팅
- keyboardActions : onDone 삭제
- 검색바 프리뷰 수정
- 오디오태그 width 수정
- 배너 width 수정

vahkjsdf committed:
[mod/#8] SearchBookItem 패딩 수정

+ lint format

vahkjsdf committed:
[mod/#8] 오디오북 분기 처리

vahkjsdf committed:
[mod/#8] 완독률 텍스트 패딩 수정

vahkjsdf committed:
[mod/#8] 어노테이션 및 파라미터명 추가

nhyeonii commented on app/src/main/java/sopt/org/millie/presentation/search/component/book/SearchBookItem.kt:
```diff
@@ -0,0 +1,76 @@
+    minute: String,
+    modifier: Modifier = Modifier,
+    isAudioEnabled: Boolean = true,
+    onClickBookItem: () -> Unit = {},
```
```suggestion
    onClickBookItem: () -> Unit,
```
얘 onClickBookItem은 기본값을 주기보다 onClickBookItem: () -> Unit 형태로 필수 파라미터로 받는 게 더 좋아 보여용 !! 클릭 이벤트는 UI 내부에서 자체적으로 처리할 일이 거의 없고, 보통 상위 ViewModel 또는 상위 스크린에서 실제 비즈니스 로직을 처리하기 때문이에요이 !!  필수로 강제하면 클릭 로직 누락을 예방할 수 있어서 컴포넌트 재사용성도 더 안전해져용 ~~

vahkjsdf replied:
넵! 알겠습니당ㅎㅎ

sohee6989 commented on app/src/main/java/sopt/org/millie/presentation/search/component/book/SearchWandokRate.kt:
```diff
@@ -0,0 +1,65 @@
+            text = rate,
+            color = MillieTheme.colors.gray3,
+            style = MillieTheme.typography.body.body1,
+        )
```
요기도 다른데 해주는 김에 한 줄 간격 주자용~~ㅎㅎ

sohee6989 reviewed (approved):
다 잘 반영된 것 같아요!!
뱃지 부분 컴포넌트 시킬거면 그 부분만 해주면 될 듯 합니다!!
수고하셨습니다👍👍

nhyeonii commented on app/src/main/java/sopt/org/millie/presentation/search/component/book/SearchWandokRate.kt:
```diff
@@ -0,0 +1,65 @@
+            tint = MillieTheme.colors.gray3,
+        )
+
+        Spacer(modifier = Modifier.height(2.dp))
```
Spacer는 부모 레이아웃의 방향을 따라서 height라고 줘도 문제 없지만, Row 안에서는 보통 width로 가로 간격을 조정 해요 !
```suggestion
        Spacer(modifier = Modifier.width(2.dp))
```

vahkjsdf replied:
아 spacer은 height만 사용하는 줄 알았네용..ㅎㅎ 수정하겠습니다!

nhyeonii commented on app/src/main/java/sopt/org/millie/core/designsystem/component/SearchTextField.kt:
```diff
@@ -0,0 +1,109 @@
+    value: String,
+    onValueChange: (String) -> Unit,
+    modifier: Modifier = Modifier,
+    onCancelClick: () -> Unit = { onValueChange("") },
```
```suggestion
    onCancelClick: () -> Unit,
```
물론 지금 코드도 문제 없지만 onCancelClick 내부에서 value를 직접 초기화하는 것보다 상위에서 함께 처리할 수 있도록 이벤트만 외부로 넘기면 더 재사용성이 높아질 것 같은데 어떻게 생각하시나용 ?! 근데 지금처럼 작성하는것두 갠찮아요 !! 이건 진짜 지민이 의견대루 하면댐 !!

vahkjsdf replied:
아 초기화 로직을 아예 상위로 넘기자는 말씀이시죠?? 좋아요!!

sohee6989 replied:
이거 근데 저 궁금한거 하나 있습니다..!!
그 저 취소 아이콘이 항상 나오는게 아닌데 이거 필수 파라미터로 남겨놓으면
취소 아이콘이 안 나타나는 상황에서 이 클릭 처리도 빈 람다로 놔둬야 하는거 아닌가용?? 
그러면 이렇게 기본값 두는 것도 좋은 것 같다고 생각하는데 이것 관련해서 의견이 궁금합니다!

`onCancelClick: () -> Unit = {}`
그래서 요렇게 하는건 어떠세용??

nhyeonii replied:
디쌤들께 여쭤봣을때 취소 아이콘은 바코드 아이콘이 잇든 없든 텍스트를 치면 항상 나온다고 하셧어요 !! 그래서 `onCancelClick: () -> Unit,` 이런식으로 작성하고 상위에서 처리해도 될 것 같다구 말씀드린 거엿어용 !!  소희 언니가 말해준대루 빈 람다로 기본값 두는 것두 괜찮구요 !!  사용 편의성을 위해서는 기본값 = {}를 주는 것도 좋을 것같아요잉  왜냐하면 대부분의 경우 그냥 텍스트만 지우면 됨이니까요!! 지민이 편한대루 하면 될것 같아용 !!

vahkjsdf replied:
입력 내용이 있으면 취소 아이콘이 항상 존재하고 이 아이콘을 클릭했을 때 입력값이 삭제되는건 필수 상황이라고 생각해서 필수 파라미터로 남겨뒀습니당

nhyeonii reviewed (commented):
넘 굿 입니다 ~~~ 호이스팅 관련 사항만 수정하면 더더 조을듯 !!

vahkjsdf committed:
[mod/#8] onCancelClick 상태 호이스팅

vahkjsdf committed:
[mod/#8] 뱃지 컴포넌트 분리

vahkjsdf committed:
[mod/#8] 완독률, 완독시간 타입 수정

vahkjsdf committed:
[add/#8] BannerColor 추가

vahkjsdf committed:
[feat/#8] SearchBanner 수정

vahkjsdf committed:
[mod/#8] lint format

vahkjsdf committed:
Merge branch 'develop' of https://github.com/SOPT-all/37-COLLABORATION-ANDROID-MILLIE into feat/#8-search-component

nhyeonii reviewed (approved):
LGTM ~ 🚀 넘 잘햇다 고생햇어용 !!

nhyeonii commented on app/src/main/java/sopt/org/millie/presentation/search/component/book/SearchBanner.kt:
```diff
@@ -0,0 +1,73 @@
+                color = MillieTheme.colors.darkGray1,
+                style = MillieTheme.typography.title.subHead4,
+            )
+
```
<img width="609" height="204" alt="image" src="https://github.com/user-attachments/assets/ae2d61df-e8b1-4e30-b12a-56aeb30be682" />
엇 근데 여기 텍스트랑 텍스트 사이 10.dp 빠진 것 같다 !! 이것만 확인 부탁해용

vahkjsdf committed:
[mod/#8] bookImage customshadow 추가

nhyeonii commented on app/src/main/java/sopt/org/millie/presentation/search/component/book/SearchBanner.kt:
```diff
@@ -0,0 +1,73 @@
+        ) {
+            Text(
+                text = bannerTitle,
+                modifier = Modifier.padding(top = 21.dp),
```
요 패딩 상위 컬럼에서 적용해도 괜찮을 것 같아요 !!.padding(top = 21.dp, bottom = 23.dp, end = 16.dp), 이렇게 위에 추가해도 될듯합니동

vahkjsdf committed:
[mod/#8] searchbanner 패딩 수정

nhyeonii reviewed (approved):
LGTM ~~~ 🤩🚀

sohee6989 commented on app/src/main/java/sopt/org/millie/presentation/search/component/book/SearchWandokRate.kt:
```diff
@@ -0,0 +1,67 @@
+        Spacer(modifier = Modifier.width(2.dp))
+
+        Text(
+            text = "$rate%",
```
이건 코드리뷰 반영 안 해도 되는 건데 그냥 하나 남겨놓는 건데요..!

문자열 템플릿을 사용할 때 변수명을 ${변수}처럼 중괄호로 감싸는 방식이
식별 오류를 줄이고 템플릿 안에서 변수를 사용했다는 점도 더 명확하게 보여준다고 합니다.
개인적으로도 가독성이 좋아서 그렇게 사용하려고 습관을 들이고 있고요!
참고 해주시면 좋을 것 같아 코멘트 남겨 놓습니다~ㅎ.ㅎ

vahkjsdf replied:
저 궁금한게.. 이 부분을 ${rate}로 해도 lint format을 하면 저렇게 바뀌는데.. 그냥 둬도 괜찮겠죠..?ㅎㅎ

sohee6989 replied:
허거덩.... 왜 그러지..??ㅎㅎ
그럼 그냥 둬야 될 것 같네요!!ㅎ.ㅎ

nhyeonii replied:
허걱 이거 내가 확인해보고 수정 필요하면 수정해둘게요 !!!!!!!

sohee6989 commented on app/src/main/java/sopt/org/millie/presentation/search/component/book/SearchBookItem.kt:
```diff
@@ -0,0 +1,86 @@
+                .fillMaxWidth()
+                .height(150.dp)
+                .customShadow(
+                    shape = RectangleShape,
```
여기 지금 그냥 사각형이 아니라 top-right, bottom-right에 radius 들어간 사각형이어서 확인 한번만 부탁드려요!!

sohee6989 commented on app/src/main/java/sopt/org/millie/presentation/search/component/book/SearchBanner.kt:
```diff
@@ -0,0 +1,76 @@
+            contentDescription = null,
+            modifier = Modifier
+                .size(width = 64.dp, height = 70.dp)
+                .align(Alignment.Bottom),
```
요기 배너도 shadow 들어가고 top-right에 radius 들어가는 것 같아요..!!

sohee6989 reviewed (approved):
여기 shadow 관련 코멘트 2개만 반영 부탁드립니다!!

vahkjsdf committed:
[mod/#8] shadow 코리 반영

nhyeonii reviewed (approved):
LGTM ~~
